app/controllers/purchases_controller.rb

```diff
@@ -22,9 +26,16 @@ def create
     end
   end
 
+  def limit
+    @@purchase_limit = params[:commit].to_i
```

Comment:
As you mentioned, changing the params to e.g. `:limit` will really help readability:)

Reply:
Yep! I'd actually changed this in the other branch as I'd switched from f.submit to f.button to give me more tools to get the date buttons to work, and I swapped over the other buttons in the partial to the same format.